miner/scroll_worker.go

```diff
@@ -1181,3 +1181,31 @@ func (w *worker) handleReorg(trigger *reorgTrigger) error {
 func (w *worker) isCanonical(header *types.Header) bool {
 	return w.chain.GetBlockByNumber(header.Number.Uint64()).Hash() == header.Hash()
 }
+
+// CalculateBlockDeadline calculates the deadline for block production based on
+// SystemContract configuration and current header information.
+// This function abstracts the deadline calculation logic for easier testing.
+func CalculateBlockDeadline(config *params.SystemContractConfig, header *types.Header) time.Time {
+	period := config.Period
+	if period == 0 {
+		period = 1 // Default to 1 second period
+	}
+
+	blocksPerSecond := system_contract.CalcBlocksPerSecond(config.BlocksPerSecond)
+	periodMs := system_contract.CalcPeriodMs(config.BlocksPerSecond)
```

Comment:
```suggestion
	periodMs := system_contract.CalcPeriodMs(blocksPerSecond)
```

Reply:
fix in [00f37de](https://github.com/scroll-tech/go-ethereum/pull/1174/commits/00f37de08a1a80ef409e51813e549af27e36dbf6)